network/Message.cpp

```diff
@@ -85,15 +94,74 @@ char* Message::Data() noexcept
 void Message::Swap(Message& rhs) noexcept {
     std::swap(m_type, rhs.m_type);
     std::swap(m_message_size, rhs.m_message_size);
+    std::swap(m_is_compressed, rhs.m_is_compressed);
     std::swap(m_message_text, rhs.m_message_text);
 }
 
 void Message::Reset() noexcept {
     m_type = MessageType::UNDEFINED;
     m_message_size = 0;
+    m_is_compressed = false;
     m_message_text.clear();
 }
 
+void Message::Compress() noexcept {
+    if (!m_is_compressed) {
+        std::string_view::size_type temp = m_message_size;
+        m_message_text = CompressImpl(m_message_text);
+        m_message_size = m_message_text.size();
+        m_is_compressed = true;
+    }
+}
+
+std::string Message::CompressImpl(const std::string& data) {
+    boost::iostreams::basic_array_source<char> src(data.c_str(), data.size());
+    boost::iostreams::stream<boost::iostreams::basic_array_source<char>> s_src(src);
+
+    std::string compressed;
+    compressed.reserve(std::pow(2.0, 26.0));
+
+    boost::iostreams::back_insert_device<std::string> cmp(compressed);
+    boost::iostreams::stream<boost::iostreams::back_insert_device<std::string>> s_cmp(cmp);
+
+    boost::iostreams::filtering_ostreambuf out;
+    out.push(boost::iostreams::zlib_compressor());
+    out.push(boost::iostreams::base64_encoder());
```

Comment:
Does base64 necessary?

Reply:
probably only if it will be wrapped in another layer of XML, which I guess doesn't apply in this case

Reply:
Removed the base64 encoding and decoding in https://github.com/freeorion/freeorion/pull/3854/commits/4ead6795ce2b51bce839955a679750840ada4831.